crates/sc-proof-of-time/src/pot_state.rs

```diff
@@ -57,6 +58,16 @@ enum ExtendAction {
     NoMerge,
 }
 
+/// Summary of the current state.
+#[derive(Debug, Clone)]
+pub struct PotStateSummary {
+    /// Current tip.
+    pub tip: String,
```

Comment:
Why do you need a string here? You could just store `Some(SlotNumber)` and `Debug` implementation will have no issues printing it when needed, but you'll save a bunch of unnecessary allocations otherwise.